repartitions/algorithm.py

```diff
@@ -0,0 +1,200 @@
+from typing import List
```

Comment:
So, my comments on these files are only about the typos, and this is because I just cannot understand the code.

I completely trust you (and the tests) about the algorithm, but if you want to give your own code a chance to not being dropped by the future maintainers of this project, please add comprehensive docstrings (which also explain the variables) and some comments; do not hesitate to be talkative.

At a first glance, we cannot even understand what is the main function, let alone the various `i`, `j`, `s`… (maybe there is not a better name for them, in that case explain thoroughly what the loop does before).

Thanks